clients/appointments/src/components/index.tsx

```diff
@@ -87,7 +86,7 @@ export const AppointmentsView = ({
       onError: handleError,
       setLoading,
       appointmentCoding: {
-        code: getAppointmentType(appointmentCancellation.appointment.code),
```

Comment:
I looked up other uses of `getAppointmentType` and it is only used for display purposes outside of this.

Reply:
Could `appointmentCancellation.appointment.code` every be undefined or null in that chain?

Reply:
Maybe throw exception if it's undefined because we can't recover right?

Reply:
sounds good. I can do that

Reply:
Actually, if it doesn't exist, what `getAppointmentType` is default to `defaultAppointmentType`:

```
export const defaultAppointmentType = {
  type: 'Office Visit',
  code: '308335008',
}
```

So should we just do option chaining then `|| defaultAppointmentType.code`?

Reply:
I don't know if not matching the code will cause the API request to error or not.

Reply:
Yea let's do that. I am going to keep the error handling as it captures more than just the destructure on this line.

Reply:
> I don't know if not matching the code will cause the API request to error or not.

It will cause the API to error. However, the user will now see the same behavior either way. If we add the default, it allows for the appointments embed to at least try to cancel. If that call fails, we show the `Error Cancelling Appointments` view.